docs: add basic-auth guides (#531)
kubeshop/kusk-gateway: docs/guides/basic-auth.md

# Authentication

> Currently the methods of authentication you can use are any methods that are Header based. That includes Basic authentication, JWT tokens, Bearer tokens, etc.

Reviewer: ```suggestion
Currently you can use any Header based authentication methods. This includes:
- Basic Authentication
- JWT Tokens
- Bearer Tokens
```

> To use the authentication, you will need to add the following section to your `x-kusk`:

Reviewer: ```suggestion
To set up authentication, add the following section to your `x-kusk` settings block:
```

```yaml
openapi: 3.0.0
info:
  title: simple-api
  version: 0.1.0
x-kusk:
  auth:
    type: basic
    auth-upstream:
      host:
        hostname: basic-auth-svc.default
        port:8080
```

The example above authenticates requests to the whole API.

> You can also specify different authentication settings for a specific operation or path. The following example shows authentication configuration for a specific operation:

Reviewer: ```suggestion
You can also specify different authentication settings for specific operations or paths. The following example shows an example authentication configuration for a specific operation:
```

```yaml
...
paths:
  /hello:
    get:
      operationId: getHello
      auth:
        type: basic
        auth-upstream:
          host:
            hostname: basic-auth-svc.default
            port:8080
      ..
```

> As you can see in the example, in the `hostname` section we're referencing a service in our cluster that will handle the authentication of the requests. This flow looks like the following: 

Reviewer: ```suggestion
In the example, in the `hostname` section we're referencing an upstream service in our cluster that will handle request authentication. The flow is as follows: 
```
